### ISSUE TITLE
Check whether function has return typehint

### ISSUE THREAD

**Issue by CZechBoY** [phpstan/phpstan-strict-rules#14]
Hi,
I would like to report methods which return type is not known - no php typehint/phpdoc.

### Summary of a problem or a feature request

I know that there is rule that checks whether function returns which it typehints - eg. 
```php
function a(): string
{
    return new stdClass; 
}
```
this is correctly reported as `should return string but returns stdClass`.
But when no typehint is defined I would expected that function "returns" `void`.

### Code snippet that reproduces the problem
https://phpstan.org/r/10e7cc35381dcada1f8b1f54095fb0b1

### Expected output
`should return void but returns stdClass`

**Comment by Majkl578** [comment]
> But when no typehint is defined I would expected that function "returns" void.

This assumption is wrong. _Nothing_ means _mixed or void_.

**Comment by CZechBoY** [comment]
@ondrejmirtes I will try at and let you know at the weekend how it goes.

**Comment by Majkl578** [comment]
Reporting missing types is fine, but the quoted assumption is incorrect.

Either way I think this rule is problematic. There are lots of occurences where type cannot be added due to BC implications (consider Doctrine for example, or just any project with BC constraints).
(In the end this is more suitable as simply Coding Standard rule, which is already in place in [Slevomat CS](https://github.com/slevomat/coding-standard/#slevomatcodingstandardtypehintstypehintdeclaration-).)

**Comment by ondrejmirtes** [comment]
> cannot be added due to BC implications

1) No one is forced to use this rule.
2) You can avoid BC issues with adding just a phpDoc.

> In the end this is more suitable as simply Coding Standard rule

I can agree with that, but this is also closely related to static analysis - more typehints you have, the better the analysis can be.

**Comment by Majkl578** [comment]
> more typehints you have, the better the analysis can be

Yes, but I don't see how it fits to this extension, which describes itself as:
> _This repository contains additional rules that revolve around strictly and strongly typed code with no loose casting for those who want additional safety in extremely defensive programming_

I'd argue that enforcing typehints is not something SA tool should be doing, that is beyond _static analysis_. (I am happily using CS rule for that.)

> No one is forced to use this rule.

This is really PITA since it's not possible to exclude rules - adding and maintaining set of all rules minus one in all consumers is slightly messy. :(

**Comment by ondrejmirtes** [comment]
I'd argue that adding omitted typehints leads to additional safety.

**Comment by Majkl578** [comment]
_Additional safety_? Maybe. _Additional safety in extremely defensive programming_? Likely not.

Also not related in terms of _no loose casting_.

**Comment by ondrejmirtes** [comment]
The definition in the README can of course change if you think it's ambiguous.

**Comment by CZechBoY** [comment]
I was surprised too that this is not in any coding-standard that I know/use.

> more typehints you have, the better the analysis can be.

This was exactly why I created this issues
